From 440302137c565a6699aa86c29c9b63c98d530703 Mon Sep 17 00:00:00 2001
From: yhe19 <yunan.he@intel.com>
Date: Mon, 10 Dec 2012 02:54:49 +0800
Subject: Dalvik GC: Change GC parameters for performance.

BZ: 47554

Change the GC heap size extension algorithm for performance.
Improve it with a heuristic heap extension algorithm in future.

Category: device-enablement
Domain: AOSP-Dalvik-GC
Origin: internal
Upstream-Candidate: no, proprietary

Change-Id: Ie170ff67e4091dbdaed8525a616d8e9b692b87bb
Orig-MCG-Change-Id: I4ad5f0e6f4757ff3b530bafd9b3892d81227b104
Signed-off-by: yhe19 <yunan.he@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/alloc/HeapSource.cpp |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/vm/alloc/HeapSource.cpp b/vm/alloc/HeapSource.cpp
index 318696e..1b58ab2 100644
--- a/vm/alloc/HeapSource.cpp
+++ b/vm/alloc/HeapSource.cpp
@@ -1460,6 +1460,12 @@ void dvmHeapSourceGrowForUtilization()
     size_t currentHeapUsed = heap->bytesAllocated;
     size_t targetHeapSize = getUtilizationTarget(hs, currentHeapUsed);
 
+    /* Add additional heap size(minFree) to the targetHeapSize to
+     * improve the performance. Improve it with a heuristic heap
+     * extension algorithm in future.
+     */
+    targetHeapSize += hs->minFree;
+
     /* The ideal size includes the old heaps; add overhead so that
      * it can be immediately subtracted again in setIdealFootprint().
      * If the target heap size would exceed the max, setIdealFootprint()
-- 
1.7.4.1

